New test : Verify nobarrier disk mount for Azure data disks

senugala opened the pull request.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/LIS/LISAv2?pullRequest=231) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/LIS/LISAv2?pullRequest=231) before we can accept your contribution.<br/><br/>:x: senugala [sign now](https://cla.opensource.microsoft.com/LIS/LISAv2?pullRequest=231)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/LIS/LISAv2?pullRequest=231) it.</sub>

iamshital commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,102 @@
+	echo "${1}" > ${HOMEDIR}/state.txt
+}
+
+CreateRAID0()
```
I think CreateRaid function is already there in utils. Please check and confirm and use it, if available.

iamshital commented on Testscripts/Windows/BVT-VERIFY-NOBARRIER.ps1:
```diff
@@ -0,0 +1,86 @@
+
+    try 
+	{
+		$count = 0
```
Fix the indents please

iamshital commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,102 @@
+ICA_TESTABORTED="TestAborted"      # Error during the setup of the test
+ICA_TESTFAILED="TestFailed"        # Error occurred during the test
+touch $STATE_FILE
+
```
Remove extra lines.

iamshital commented:
@senugala Fix all the indents and submit the results here.

iamshital commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,101 @@
+	echo "${mdVolume}	${mountDir}	ext4,nobarrier	defaults	0 0" >> /etc/fstab
+	LogMsg "INFO: Successfuly Added fstab entry"
+	UpdateTestState ICA_TESTCOMPLETED
+	
```
Remove this extra line.

iamshital commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,101 @@
+mountDir="/data"
+cd ${HOMEDIR}
+
+install_fio
```
Why you need install_fio?

iamshital commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,101 @@
+
+CreateRAID0()
+{	
+	
```
Remove this line

iamshital commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,101 @@
+# GNU General Public License for more details.
+# 
+#
+# Sample script to run sysbench.
```
The description is wrong

chvalean commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,101 @@
+# must log its output to a different directory.
+#
+
+HOMEDIR="/home/lisa/"
```
do not hardcode paths, you can use instead `pwd`

chvalean commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,101 @@
+#
+
+HOMEDIR="/home/lisa/"
+LogMsg()
```
Logmsg already exists in utils, no need to re-define here

chvalean commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,101 @@
+	fi
+	
+	LogMsg "INFO: adding fstab entry"
+	echo "${mdVolume}	${mountDir}	ext4,nobarrier	defaults	0 0" >> /etc/fstab
```
why add to /etc/fstab if this is not used further in the code?
You can use directly the mount command if such, no need to go through fstab file just to later do a simple mount

chvalean commented on Testscripts/Linux/nobarrier.sh:
```diff
@@ -0,0 +1,101 @@
+	sleep 1
+	mount -o nobarrier ${mdVolume} ${mountDir}
+	if [ $? -ne 0 ]; then
+		LogMsg "Error: Unable to create raid"
```
this message would go after the mdadm command, that will verify actually if the RAID was created or not.

juhlee-microsoft commented:
@senugala Please fill out description field in PR.

lizzha commented on XML/TestCases/BVT.xml:
```diff
@@ -382,4 +382,14 @@
         <Tags>network</Tags>
     </test>
+    <test>
+        <testName>BVT-VERIFY-NOBARRIER</testName>
```
please come out a proper case name

iamshital commented:
@senugala Please address the comments on the PR.

senugala commented:
```
Execution is verified after addressing all comments.

10/15/2018 21:33:28 : [INFO ] CleanUP Successful for ICA-RG-SingleVM-SEU1604-DJNV-636752550775..
10/15/2018 21:33:28 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-DISK-WITH-NOBARRIER~~~~~~~~~~
10/15/2018 21:33:28 : [INFO ] CURRENT - PASS    - 1
10/15/2018 21:33:28 : [INFO ] CURRENT - FAIL    - 0
10/15/2018 21:33:28 : [INFO ] CURRENT - ABORTED - 0
10/15/2018 21:33:28 : [INFO ] Checking background cleanup jobs..
10/15/2018 21:33:28 : [INFO ] Removed 0 context files.
10/15/2018 21:33:28 : [INFO ] Cycle Finished.. TC-17832
10/15/2018 21:33:28 : [INFO ]
Test Results Summary
ICA test run on 10/15/2018 21:21:06
ARM Image under test Canonical : UbuntuServer : 16.04-LTS : latest
Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:12
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VERIFY-DISK-WITH-NOBARRIER : PASS : 12.29
        Standard_DS1_v2 : Number of Disk Attached - 4 : PASS


Logs can be found at .\TestResults\2018-16-10-02-50-57-6415


10/15/2018 21:33:28 : [INFO ] Creating 'Azure-17832-buildlogs.zip' from '.\TestResults\2018-16-10-02-50-57-6415'
10/15/2018 21:33:28 : [INFO ] F:\Git-Hub\LISAv2\Azure-17832-buildlogs.zip created successfully.
10/15/2018 21:33:28 : [INFO ] Copied : .\report\report_TC-17832.xml --> .\report\report_TC-17832-junit.xml
10/15/2018 21:33:28 : [INFO ] Analysing results..
10/15/2018 21:33:28 : [INFO ] PASS  : 1
10/15/2018 21:33:28 : [INFO ] FAIL  : 0
10/15/2018 21:33:28 : [INFO ] ABORT : 0
10/15/2018 21:33:28 : [INFO ] LISAv2 exits with code: 0

```

iamshital commented on Testscripts/Windows/VERIFY-DISK-WITH-NOBARRIER.ps1:
```diff
@@ -0,0 +1,78 @@
+
+    try {
+        $count = 0
+        foreach ($VM in $allVMData) {
```
Remove this foreach loop and assing $VM = $AllVMData

iamshital commented on XML/TestCases/FunctionalTests.xml:
```diff
@@ -1159,4 +1159,20 @@
+        <Platform>Azure</Platform>
+        <Category>Functional</Category>
+        <Area>STORAGE</Area>
+        <Tags>storage</Tags>
```
rename : storage --> disk

iamshital replied:
and add one more tag : hv_storvsc

lizzha commented on Testscripts/Linux/utils.sh:
```diff
@@ -2836,14 +2836,16 @@ function remove_partitions () {
 		local deviceName="/dev/md1"
 		local mountdir=/data-dir
 		local format="ext4"
+		local mount_option=""
```
when $1, $2 $3 is specified, but $4 not provided, it goes to else branch, which is not expected

iamshital replied:
I think this is expected operation here. when $4 is not given, it goes to else block and assigns empty value to mount_option. Later, following code is executed.
```
	if [ -z "$mount_option" ]
	then
		mount $deviceName $mountdir
	else
		mount -o $mount_option $deviceName $mountdir
	fi
```

lizzha replied:
when $4 is not given, but $1, $2 $3 are given, in current logic, the given $1 $2 and $3 will not be set properly

iamshital replied:
Got you. @senugala Please remove the hard coded values from else block and assign correctly.
Thank you @lizzha for making code better.

lizzha replied:
Suggest to make the change like:
local deviceName="/dev/md1" 
local mountdir=/data-dir 
local format="ext4" 
local mount_option=""

if [ ! -z $1 ]; then
	deviceName=$1
fi
if [ ! -z $2 ]; then
	mountdir=$2
fi
if [ ! -z $3 ]; then
	formant=$3
fi
if [ ! -z $4 ]; then
	mount_option=$4
fi

senugala replied:
Thank you Lizzha and everyone. I have updated scripts based on your suggestion .

chvalean commented:
@lizzha please approve and merge this PR

lizzha commented:
I will approve once the left comment is resolved, @senugala please address the comment

senugala commented:
```
Execution is verified after addressing the all comments.

10/17/2018 19:06:41 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-DISK-WITH-NOBARRIER~~~~~~~~~~
10/17/2018 19:06:41 : [INFO ] CURRENT - PASS    - 1
10/17/2018 19:06:41 : [INFO ] CURRENT - FAIL    - 0
10/17/2018 19:06:41 : [INFO ] CURRENT - ABORTED - 0
10/17/2018 19:06:41 : [INFO ] Checking background cleanup jobs..
10/17/2018 19:06:41 : [INFO ] Removed 0 context files.
10/17/2018 19:06:42 : [INFO ] Cycle Finished.. TC-69528
10/17/2018 19:06:42 : [INFO ]
Test Results Summary
ICA test run on 10/17/2018 18:56:18
ARM Image under test Canonical : UbuntuServer : 16.04-LTS : latest
Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:10
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VERIFY-DISK-WITH-NOBARRIER : PASS : 10.32
        Standard_DS1_v2 : Number of Disk Attached - 4 : PASS


Logs can be found at .\TestResults\2018-18-10-00-26-09-5391


10/17/2018 19:06:42 : [INFO ] Creating 'Azure-69528-buildlogs.zip' from '.\TestResults\2018-18-10-00-26-09-5391'
10/17/2018 19:06:42 : [INFO ] F:\Git-Hub\LISAv2\Azure-69528-buildlogs.zip created successfully.
10/17/2018 19:06:42 : [INFO ] Copied : .\report\report_TC-69528.xml --> .\report\report_TC-69528-junit.xml
10/17/2018 19:06:42 : [INFO ] Analysing results..
10/17/2018 19:06:42 : [INFO ] PASS  : 1
10/17/2018 19:06:42 : [INFO ] FAIL  : 0
10/17/2018 19:06:42 : [INFO ] ABORT : 0
10/17/2018 19:06:42 : [INFO ] LISAv2 exits with code: 0

```
